File: src/1438-Longest-Continuous-Subarray-With-Absolute-Diff-Less-Than-or-Equal-to-Limit/1438.py
class Solution:
    def longestSubarray(self, nums: List[int], limit: int) -> int:
        maxd, mind, i = [], [], 0
        
        for a in nums:
            while maxd and a > maxd[-1]: maxd.pop()
            while mind and a < mind[-1]: mind.pop()
            maxd.append(a)
            mind.append(a)
            if maxd[0] - mind[0] > limit:
                if maxd[0] == nums[i]: maxd.pop(0)
                if mind[0] == nums[i]: mind.pop(0)
                i += 1
        return len(nums) - i

File: src/1438-Longest-Continuous-Subarray-With-Absolute-Diff-Less-Than-or-Equal-to-Limit/1438.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums, int limit) {
        deque<int> maxd, mind;
        int i = 0;
        
        for (int a : nums) {
            while (maxd.size() && a > maxd.back()) maxd.pop_back();
            while (mind.size() && a < mind.back()) mind.pop_back();
            maxd.emplace_back(a);
            mind.emplace_back(a);
            
            if (maxd[0] - mind[0] > limit) {
                if (maxd[0] == nums[i]) maxd.pop_front();
                if (mind[0] == nums[i]) mind.pop_front();
                i++;
            }
        }
        return nums.size() - i;
    }
};